formats/custom/stub.go

```diff
@@ -0,0 +1,20 @@
+package custom
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type Log interface {
+	String() (string, float64)
+	Labels() prometheus.Labels
+}
+
+// Formats returns supported template formats for a given type
+func Formats() map[string][]string {
+	return map[string][]string{}
```

Comment:
I think it would be a great feature if we could add new formats without touching actual source code (without requiring to add it manually to a format hashtable, etc.).

My original idea was the following:
We declare a virtual embedded "file system" (`embed.FS`) `TemplateFS` for each `formats` folder: web, syslog, custom:

https://github.com/kube-logging/log-generator/blob/41bea61ff955416cde88aa03d3d88e89b1359a8d/formats/syslog/syslog.go#L27-L28

`Go embed` automatically populates this FS instance with `.tmpl` files.
This way, we don't have to manually keep track of our templates in code; once a new `.tmpl` file is added to the given format folder, it will be automatically compiled into the binary, and the following function can even query the names of all the available formats:

https://github.com/kube-logging/log-generator/blob/41bea61ff955416cde88aa03d3d88e89b1359a8d/formats/log.go#L96-L104

I think we can do the same with the new `custom` format folder as well and that way we could avoid building a "database" manually in Formats(), maybe we could also get rid of the `LogFactory` function too.

What do you think?

(`custom.TemplateFS` would be the "communication interface" between loggen and our custom module then.)

Reply:
TLDR: `Formats()` can be implemented using `embed.FS`, we don't need to rely on it as an interface.